src/java.base/share/classes/java/lang/invoke/VarHandles.java

```diff
@@ -629,6 +623,13 @@ private static void noCheckedExceptions(MethodHandle handle) {
                     noCheckedExceptions((MethodHandle) arg);
                 }
             }
+        } else {
+            exceptionTypes = MethodHandles.checkedExceptions(handle);
+        }
+        if (exceptionTypes != null) {
+            if (Stream.of(exceptionTypes).anyMatch(VarHandles::isCheckedException)) {
+                throw newIllegalArgumentException("Cannot adapt a var handle with a method handle which throws checked exceptions");
+            }
```

Comment:
Minimize the changes to this code to allow for easier future merges.
```suggestion
        } else {
            // Support OpenJ9 MethodHandle implementation
            Class<?>[] exceptionTypes = MethodHandles.checkedExceptions(handle);
        
            if (exceptionTypes != null) {
                if (Stream.of(exceptionTypes).anyMatch(VarHandles::isCheckedException)) {
                    throw newIllegalArgumentException("Cannot adapt a var handle with a method handle which throws checked exceptions");
                }
            }
       }
```
And the rest of the changes to this method can be removed